### PR TITLE
Support QDQ format for weight-only quantization

### PR DESCRIPTION
**Pull request by mengniwang95**
## Type of Change

feature

## Description

Support QDQ format for weight-only quantization

It requires:
- onnxruntime >= 1.19.0
- opset_version of model >=21
- quantized bits in [4, 8]

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: onnx_neural_compressor/algorithms/weight_only/gptq.py [hunk 1]
```diff
@@ -318,24 +324,50 @@ def gptq_quantize(
             )
 
             weight_tensor = model.get_initializer(node.input[1])
+            org_shape = weight.shape
             init_share_num = model.get_initializer_share_num(node.input[1])
 
-            satisfy_MatMulNBits_condition = Version(ort.__version__) > constants.ONNXRT1161_VERSION and num_bits == 4
+            satisfy_MatMulNBits_condition = ort_version > constants.ONNXRT1161_VERSION and num_bits == 4
```
Comment (thuang6) [inline onnx_neural_compressor/algorithms/weight_only/gptq.py:330]: it is better to put such a ORT version check for both MatMulNBits and standard DQ out of algorithm specific logic/file.  maybe in high level API layer by detecting quant_format

File: onnx_neural_compressor/algorithms/weight_only/rtn.py [hunk 2]
```diff
@@ -99,21 +103,46 @@ def rtn_quantize(
 
             weight = quant_utils.pad_tensor(weight, group_size, k_blocks)
 
-            satisfy_MatMulNBits_condition = (
-                version.Version(ort.__version__) > constants.ONNXRT1161_VERSION and num_bits == 4
-            )
+            satisfy_MatMulNBits_condition = ort_version > constants.ONNXRT1161_VERSION and num_bits == 4
             satisfy_MatMulFpQ4_condition = (
-                version.Version(ort.__version__) >= constants.ONNXRT116_VERSION and num_bits == 4 and group_size == 32
+                ort_version >= constants.ONNXRT116_VERSION and num_bits == 4 and group_size == 32
```
Comment (thuang6) [inline onnx_neural_compressor/algorithms/weight_only/rtn.py:108]: similar comments, it is better to put such a ORT version check for both MatMulNBits and standard DQ out of algorithm specific logic/file. maybe in high level API layer by detecting quant_format

File: onnx_neural_compressor/algorithms/weight_only/rtn.py [hunk 3]
```diff
@@ -99,21 +103,46 @@ def rtn_quantize(
 
             weight = quant_utils.pad_tensor(weight, group_size, k_blocks)
 
-            satisfy_MatMulNBits_condition = (
-                version.Version(ort.__version__) > constants.ONNXRT1161_VERSION and num_bits == 4
-            )
+            satisfy_MatMulNBits_condition = ort_version > constants.ONNXRT1161_VERSION and num_bits == 4
```
Comment (thuang6) [inline onnx_neural_compressor/algorithms/weight_only/rtn.py:106]: For quant a node logic (either MatMulBits or standard DQ), it looks quite similar for different algorithms and should be able to decoupled with algo by a good abstraction.  is it feasible to refactor?

**Comment by mengniwang95** [comment]
image

**Review by thuang6 — APPROVED** [review]

**Pull request merged**